src/core/lib/surface/call.c

```diff
@@ -1384,8 +1397,8 @@ static void receiving_initial_metadata_ready(grpc_exec_ctx *exec_ctx,
     }
   }
 
-  call->has_initial_md_been_received = true;
-  if (call->saved_receiving_stream_ready_bctlp != NULL) {
+  if (!gpr_atm_acq_cas(&call->recv_state, RECV_NONE,
```

Comment:
This uses '__ATOMIC_ACQUIRE' semantics only in case of success and '__ATOMIC_RELAXED`  semantics in case of failure.  Since we are accessing `saved_receiving_stream_ready_bctlp` only in case `gpr_atm_acq_cas` fails, we are using `__ATOMIC_RELAXED` and hence no guarantee that we are getting the updated value of `saved_receiving_stream_ready_bctlp`